doc/api/util.md

```diff
@@ -239,7 +239,47 @@ The `--throw-deprecation` command-line flag and `process.throwDeprecation`
 property take precedence over `--trace-deprecation` and
 `process.traceDeprecation`.
 
-## `util.format(format[, ...args])`
+## `util.colorize`
+<!-- YAML
+added: REPLACEME
+-->
+
+The `util.colorize` object provides functions that add ansi color codes to the
+provided string. These may be used to style terminal output.
+
+### `util.colorize.<style>[. ...<style>](string[, ...string])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `string` {string} The string that is formatted by the chosen style.
+* Returns: {string} The formatted string
+
+The API allows to be used with a builder/chaining pattern to add multiple styles
+in one call. Nesting color codes is supported.
+
+```js
+const { colorize } = util;
+
+console.log(
+  `${colorize.green('Heads up')}: only the "Heads up" is green`
+);
+
+console.log(
+  colorize.green('green', colorize.yellow('yellow'), 'green')
+)
+
+console.log(
+  colorize.bold.underline.red('bold red underline')
```

Comment:
i would definitely prefer to avoid getters; both because it's a more confusing API, and because it's slower. i'd follow colors here, not chalk.

Reply:
oh no, i meant `colors/safe` :-) definitely nobody should use the `colors` style :-)

Reply:
I am somewhat puzzled: that exact API is the one implemented here?

Reply:
oh, maybe i'm confused. i'd prefer `colorize.bold().underline()` to `colorize.bold.underline`, to be clearer.

Reply:
I did consider that as well and I am open to using that, if it's what a majority prefers. Right now I would stick to the existing due to these reasons:

1. Users of colors and chalk are used to it
2. Passing a string to only the last call might also be confusing without a `.write()`/`.format()` similar function
3. The performance should pretty much be identical and there should be room for improvement one way or the other (the implementation is already decent performance wise while chalk is still faster)

Reply:
i'm still a bit confused what the string at the end does - i've never used it in any of my chalk/colors usage.